ethers-providers/src/transports/ws.rs

```diff
@@ -140,6 +144,26 @@ impl Ws {
         Ok(Self::new(ws))
     }
 
+    /// Initializes a new WebSocket Client with authentication
+    #[cfg(not(target_arch = "wasm32"))]
+    pub async fn connect_with_auth(
```

Comment:
this makes more sense here as adding auth is a bit more verbose

Reply:
Is this point still valid? If yes, could you please be a little bit more specific?